--- a/net/minecraft/world/entity/ItemBasedSteering.java
+++ b/net/minecraft/world/entity/ItemBasedSteering.java
@@ -14,6 +_,7 @@
    private final EntityDataAccessor<Boolean> f_20839_;
    public boolean f_20834_;
    public int f_20835_;
+   public int boostTimeTotal;
 
    public ItemBasedSteering(SynchedEntityData p_20841_, EntityDataAccessor<Integer> p_20842_, EntityDataAccessor<Boolean> p_20843_) {
       this.f_20837_ = p_20841_;
@@ -67,4 +_,12 @@
    public boolean m_20851_() {
       return this.f_20837_.m_135370_(this.f_20839_);
    }
+
+   // CraftBukkit add setBoostTicks(int)
+   public void setBoostTicks(int ticks) {
+      this.f_20834_ = true;
+      this.f_20835_ = 0;
+      this.f_20837_.m_135381_(this.f_20838_, ticks);
+   }
+   // CraftBukkit end
 }
